python/src/scipp/_variable.py

```diff
@@ -300,29 +292,22 @@ def array(*,
     :seealso: :py:func:`scipp.zeros` :py:func:`scipp.ones`
               :py:func:`scipp.empty` :py:func:`scipp.scalar`
 
-    :param dims: Dimension labels.
+    :param dims: Dimension labels, nonempty.
```

Comment:
Why is empty not allowed? What if `values` points to a 0-D numpy array?

Reply:
That would make a scalar. I disallowed it to get a clearer distinction between `array` and `scalar`. In particular because `sc.array([], values=[1, 2])` makes a scalar containing a list which might be confusing.

Reply:
But a 0-D array is an array. Not sure I understand why it should be disallowed. The example you give makes sense, and I don't feel it is confusing? Yes, it may hide (rare?) typoes.

But look at it from another point of view: So you want to write a function that works with any numpy array, and you want to use `sc.array`. As it is now you cannot, since it will for if someone passes a 0-D array to the function. Would you force use of `sc.Variable` instead?

Reply:
Yes, I agree.

Combining this with the discussion on `shape` and `with_variances` below, we could remove `array` completely because it would be identical to the bare constructor.

Reply:
I would hesitate slightly before removing it, at least until we figured out how we want to support contruction of structured dtypes. We may want to support that in `sc.Variable` (such that, e.g., desiralization code is simple and generic) but not in `sc.array`. So unless you see a good reason for removal now, I would keep `array` and reconsider it later.

Reply:
Ok. I think it makes sense to keep it even if it's just to help users draw the distinction between arrays and scalars in most cases.

But do you agree that there would be no functional distinction between the constructor and `array` after these changes?

Reply:
Yes, I think this is indeed the case (for now).

Reply:
What do you think of `sc.Variable(dims=())`? It makes a default-/zero-initialised scalar. Should this be allowed?

Reply:
Seems inconsistent. I feel `values` should be a required arg?

Reply:
I think so, too.